### PR TITLE
List table names action v1

### PR DESCRIPTION
**Pull request by TrangPham**

### CODE REVIEW

File: great_expectations_cloud/agent/actions/list_table_names.py [hunk 1]
```diff
@@ -28,7 +28,7 @@ class ListTableNamesAction(AgentAction[ListTableNamesEvent]):
     @override
     def run(self, event: ListTableNamesEvent, id: str) -> ActionResult:
         datasource_name: str = event.datasource_name
-        datasource = self._context.get_datasource(datasource_name=datasource_name)
+        datasource = self._context.get_datasource(datasource_name)
```
Comment (joshua-stauffer) [inline great_expectations_cloud/agent/actions/list_table_names.py:31]: this should be updated to V1 syntax:
```suggestion
        datasource = self._context.data_sources.get(name=datasource_name)
```

File: tests/integration/actions/conftest.py [hunk 2]
```diff
@@ -72,9 +72,9 @@ def datasource(
         datasource.name == datasource_name
     ), "The datasource was not updated in the previous method call."
     yield datasource
-    context.delete_datasource(datasource_name=datasource_name)
+    context.delete_datasource(datasource_name)
```
Comment (joshua-stauffer) [inline tests/integration/actions/conftest.py:75]: here is the V1 syntax:
```suggestion
    context.data_sources.delete(name=datasource_name)
```

File: tests/integration/actions/conftest.py [hunk 3]
```diff
@@ -72,9 +72,9 @@ def datasource(
...
     with pytest.raises(get_missing_datasource_error_type):
-        context.get_datasource(datasource_name=datasource_name)
+        context.get_datasource(datasource_name)
```
Comment (joshua-stauffer) [inline tests/integration/actions/conftest.py:77]: likewise
```suggestion
        context.data_sources.get(name=datasource_name)
```

File: great_expectations_cloud/agent/actions/list_table_names.py [hunk 4]
```diff
@@ -49,12 +49,12 @@ def run(self, event: ListTableNamesEvent, id: str) -> ActionResult:
...
-                url=f"{self._base_url}/organizations/"
-                f"{self._organization_id}/datasources/{datasource_id}",
-                json={"table_names": table_names},
+            response = session.put(
+                url=f"{self._base_url}/api/v1/organizations/"
+                f"{self._organization_id}/table-names/{datasource_id}",
+                json={"data": {"table_names": table_names}},
             )
-        if response.status_code != 204:  # noqa: PLR2004
+        if response.status_code != 200:  # noqa: PLR2004
```
Comment (rreinoldsc) [inline great_expectations_cloud/agent/actions/list_table_names.py:57]: nice catch

File: tests/integration/actions/test_list_table_names.py [hunk 5]
```diff
@@ -110,5 +108,5 @@ def test_running_list_table_names_action_fails_for_unreachable_datasource(
 
     # Act & Assert
     # Check that the action was unsuccessful and an error was raised.
-    with pytest.raises(ValueError):
+    with pytest.raises(KeyError):
```
Comment (rreinoldsc) [inline tests/integration/actions/test_list_table_names.py:111]: Both these errors are surprising to see here. Where is the KeyError coming from?

**Review by rreinoldsc — APPROVED** [review]

**Pull request merged**